admin/workspace-providers/deployment.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Deployment
+description: Learn how to deploy a workspace provider.
+state: beta
+---
+
+This article walks you through the process of deploying a workspace provider to
+a [Kubernetes cluster](../../setup/kubernetes/index.md).
+
+[Workspace Providers](index.md) are logical groups of resources to which
+developers can deploy environments. As with the Coder installation, Helm manages
+the deployment of workspace providers into the Kubernetes cluster that will contain
+your environments.
+
+## Dependencies
+
+Install the following dependencies if you haven't already:
+
+- [Coder CLI](../../cli/installation.md)
+- [Helm](https://helm.sh/docs/intro/install/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Requirements
+
+1. Workspace providers **must have a hostname set** that is a subdomain of the
+   Coder deployment. For example, if the Coder deployment's hostname is
+   `coder.example.com`, the workspace provider's hostname must match the format
+   `*.coder.example.com`.
+1. The main Coder deployment and the workspace provider must be able to
+   communicate bi-directionally via their respective hostnames.
+1. The workspace provider scheme (HTTP or HTTPS) must match that of the Coder
+   deployment Access URL.
+1. The Kubernetes cluster address must be reachable from the Coder deployment.
+
+## Connecting to the Cluster
+
+To add a Kubernetes cluster as a workspace provider, you must first make sure
+that you're connected to the cluster you want to expand into. Run the following
+command:
+
+```bash
+kubectl config current-context
+```
+
+Confirm that your current kubectl context correct before continuing; otherwise,
+connect to the correct context.
+
+## Creating the Coder Namespace (Optional)
+
+We recommend running workspace providers in a separate
+[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/);
+to do so, run:
+
+```bash
+kubectl create namespace coder
+```
+
+Next, change the kubectl context to point to your newly created namespace:
+
+```bash
+kubectl config set-context --current --namespace=coder
+```
+
+## Creating the New Workspace Provider
+
+Using the Coder CLI, create a new workspace provider in the `pending` state.
+
+```bash
+coder providers create [NAME] \
+    --hostname=[HOSTNAME] \
+    --clusterAddress=[CLUSTER_ADDRESS]
+```
+
+You must provide the following arguments:
+
+- `name`: A unique name of the workspace provider
+- `hostname`: The hostname of the workspace provider
+- `clusterAddress`: The address of your Kubernetes cluster. You can retrieve this
```

Comment:
```suggestion
- `cluster-address`: The address of your Kubernetes cluster. You can retrieve this
```

There's a related suggestion above from Mike

Reply:
I think this suggestion was originally added then undone. [Confirmed again](https://github.com/cdr/coder-cli/commit/f34aafe57e79f7006ba8dc6e8bf6cb90bc9c8303#diff-7da9dc66d57a11a014bd5c611c39deab6ab38dc49dcc91d35011ef9d20457de0) it's hyphenated, so will be making this change.

Reply:
Can blame me for that - I misread Jonathan and Colin's thread. Sorry everyone

Reply:
@vapurrmaid no worries -- all is well!